File: dbaasp_api_helper_libraries/ruby/APICaller.rb
 require "./request/AbstractRequest.rb"
 require './request/LookupRequest.rb' 
 require './request/PeptideCardRequest.rb' 
 require './request/SearchRequest.rb' 
 require './request/RankingSearchRequest.rb' 
 require './LookupType.rb' 
 require './FormatType.rb' 
 require './MathOperationTypes.rb' 
 require './Complexity.rb' 
 
#  lookupRequest = LookupRequest.new
#  lookupRequest.lookupType = LookupType::KINGDOM;
#  lookupRequest.format = FormatType::XML;
#  p lookupRequest.request()


#  peptideCardRequest = PeptideCardRequest.new
#  peptideCardRequest.peptide_id = 1
#  peptideCardRequest.format = FormatType::XML;
#  p (peptideCardRequest.request())


#  searchRequest = SearchRequest.new
#  searchRequest.name = "a"
#  searchRequest.format = FormatType::JSON;
#  p (searchRequest.request())


#  rankingSearchRequest = RankingSearchRequest.new
#  rankingSearchRequest.target_species_id = 3232
#  rankingSearchRequest.activity_measure_id = 72
#  rankingSearchRequest.activity = 100;
#  rankingSearchRequest.operation = MathOperationTypes::GREATER;
#  rankingSearchRequest.format = FormatType::FASTA;
#  p (rankingSearchRequest.request())

File: dbaasp_api_helper_libraries/python/APICaller.py
import sys
sys.path.append('./request')  

from LookupRequest import LookupRequest
from PeptideCardRequest import PeptideCardRequest
from SearchRequest import SearchRequest
from RankingSearchRequest import RankingSearchRequest

from LookupType import LookupType
from FormatType import FormatType
from Complexity import Complexity
from MathOperationTypes import MathOperationTypes

# lookupRequest = LookupRequest()
# lookupRequest.lookupType = LookupType.MEDIUM;
# print (lookupRequest.request())


# peptideCardRequest = PeptideCardRequest()
# peptideCardRequest.peptide_id = 8;
# peptideCardRequest.format = FormatType.JSON;
# print (peptideCardRequest.request())


# searchRequest = SearchRequest()
# searchRequest.name = "a"
# searchRequest.format = FormatType.JSON;
# print (searchRequest.request())


# rankingSearchRequest = RankingSearchRequest()
# rankingSearchRequest.target_species_id = 3232
# rankingSearchRequest.activity_measure_id = 72
# rankingSearchRequest.activity = 100;
# rankingSearchRequest.operation = MathOperationTypes.GREATER;
# rankingSearchRequest.format = FormatType.JSON;
# print (rankingSearchRequest.request())
